src/banded/infreversecholeskytridiagonal.jl

```diff
@@ -0,0 +1,133 @@
+const MAX_TRIDIAG_CHOL_N = 2^21 - 1
+mutable struct LazySymTridiagonalReverseCholeskyFactor{T,M1,M2} <: BandedMatrices.AbstractBandedMatrix{T}
```

Comment:
Note in existing code we treat factors as a `LazyMatrix` and use the memory layout to indicate that it is banded. We also wrap a (possibly banded) cached matrix:

https://github.com/JuliaLinearAlgebra/InfiniteLinearAlgebra.jl/blob/f2fe965b5ad1901a756b0a790abade7f74b7678a/src/infcholesky.jl#L2

In the experimental QL it is made an `AbstractCachedMatrix`:

https://github.com/JuliaLinearAlgebra/InfiniteLinearAlgebra.jl/blob/f2fe965b5ad1901a756b0a790abade7f74b7678a/src/infql.jl#L374

I think the different patterns should be unified and the variable names made consistent.

Note @TSGut called it "Factors" instead of "Factor" because you access it via `cholesky(A).factors` so probably we should make the name consistent.

A possible alternative design would be to store just the data and wrap the factor in a `BandedMatrix`. Or indeed in this case `Bidiagonal`.

Reply:
I didn't realise that `LazyMatrix` was an abstract type, thanks. I left the rest for now since as you say maybe it needs a bit more thought.